core/vm/instructions.go

```diff
@@ -876,19 +881,34 @@ func opETX(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte
 	if err != nil {
 		temp.Clear()
 		stack.push(&temp)
+		fmt.Printf("%x opETX error: %s\n", scope.Contract.self.Address(), err.Error())
 		return nil, nil // following opCall protocol
 	}
 
 	sender := scope.Contract.self.Address()
+	globalNonce, err := interpreter.evm.StateDB.GetNonce(common.ZERO)
```

Comment:
I think two commits accidentally got squashed here. Half of the commit is dealing with ETX set management & spending. The other piece is the missing global ETX nonce logic. These should be in different commits.

The commit message and subject line are equally mix-matched, haha

Reply:
Not sure what you want me to do here. Do you want me to make a new commit with some of the logic?

Reply:
Yes, please split into atomic commits

Reply:
Unfortunately I can't find which commit you're referring to. Can you put the commit hash?

Reply:
Oops, idk why I thought GitHub would take you to my comment in the commit, and not the aggregate diff... sorry. Here it is: https://github.com/dominant-strategies/go-quai/pull/470/commits/54e913863b9720041f57bfdf88341464aa488903

Reply:
I believe this has been fixed